### DISCUSSION TITLE
Failing to build project and get Kloadgen jar

### DISCUSSION THREAD

**Discussion by ShantiSumanth** [sngular/kloadgen#137] [Q&A]
Hi, 
When this,"mvn clean install -P standalone" command is executed i'm  getting error "Failed to execute goal on project kloadgen: Could not resolve dependencies for project net.coru:kloadgen:jar:3.5.3: Failure to find com.github.everit-org.json-schema:org.everit.json.schema:jar:1.12.1 in http://repo-hybdev.europe.odcorp.net/nexus/repository/public/ was cached in the local repository, resolution will not be reattempted until the update interval of od-repo has elapsed or updates are forced". 

I'm new to this, Can someone help me in sorting this please ? or would appreciate if anyone can provide me kloadgen plugin

Thanks

**Comment by Rivarsal** [comment]
Hello @ShantiSumanth !

Could you try to do "mvn clean install" without the "-P standalone"? If that doesn´t work, I will be watching this question to help with your questions.

**Reply by ShantiSumanth** [reply]
Hi @Rivarsal,
Thanks for your response. Same error was thrown when ran without -P standalone as well . Please find error details below 

[WARNING] The POM for com.github.everit-org.json-schema:org.everit.json.schema:jar:1.12.1 is missing, no dependency information available
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  0.955 s
[INFO] Finished at: 2021-11-17T15:53:19Z
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal on project kloadgen: Could not resolve dependencies for project net.coru:kloadgen:jar:3.5.3: Failure to find com.github.everit-org.json-schema:org.everit.json.schema:jar:1.12.1 in http://repo-hybdev.europe.odcorp.net/nexus/repository/public/ was cached in the local repository, resolution will not be reattempted until the update interval of od-repo has elapsed or updates are forced -> [Help 1]

**Reply by ShantiSumanth** [reply]
not sure how to resolve the dependency issue. Would be great if you can help in fixing this. Thanks

**Reply by Rivarsal** [reply]
It might be an error on our end. Let me check with @jemacineiras 

Sorry for the troubles

**Reply by ShantiSumanth** [reply]
I got the plugin with jdk 17 with one of my friends help. Thanks for looking into this @Rivarsal  and for your prompt response.

**Reply by Rivarsal** [reply]
Great! That´s good news!

Thanks to this question we fixed a couple of issues we had too in our build. So thank you for bringing this up.

Also, if you need help with anything, just ask us on this forum or create a new issue.

**Reply by ShantiSumanth** [reply]
Yeah sure. Thanks @Rivarsal

**Reply by ShantiSumanth** [reply]
Hi @Rivarsal , I'm trying to produce a message to the topic but JavaRequest>kafkaSchemaSampler is not allowing me to add message.placeholder.key . Sorry i'm new to this and trying to understand, Would be great if you can provide a example script/jmx file on how to sent a message? 

Thanks

**Reply by Rivarsal** [reply]
Hello @ShantiSumanth 

First of all, I recommend downloading the master branch again since it has multiple fixes to data generation with the last version.

As for the examples, we have 2 text examples created in the root directory of the file. example-test-plan.jmx needs to have a Schema Registry API turned on. We have one API ready in our Dockerfile (with a Kafka server as well) in the folder schema_registry_docker in order to build the image you have to type the follow command in the terminal: `docker-compose -f docker-compose-noauth.yml up` and set a new schema with the Schema Registry API. 

Here is the documentation of SchemaRegistry for that: 

https://docs.confluent.io/platform/current/schema-registry/develop/api.html

Once you do that, you have to press the TestRegistry Button in the Schema Registry Config Element and choose the Schema you sent to the Schema Registry in Value Schema Load Generator Config. You can also set the Key you want to have in your messages to Kafka in Key Plain Load Generator Config

I will also give you a new file that works without schema_registry, you just need to choose the file (You need to change the type of the document back to .jmx) and choose the avro you need
[AvroTest.txt](https://github.com/corunet/kloadgen/files/7570411/AvroTest.txt)

If you need anything else, I will gladly help you out.

Pablo José López Rivadulla

**Comment by ghost** [comment]
Remove subscription

Enviado desde mi iPhone
